crates/net/network/src/session/active.rs

```diff
@@ -575,22 +578,19 @@ impl Future for ActiveSession {
                 }
 
                 // try to resend the pending message that we could not send because the channel was
-                // full.
+                // full. [`PollSender`] will ensure that we're woken up again when the channel is
+                // ready to receive the message, and will only error if the channel is closed.
                 if let Some(msg) = this.pending_message_to_session.take() {
-                    match this.to_session_manager.try_send(msg) {
-                        Ok(_) => {}
-                        Err(err) => {
-                            match err {
-                                TrySendError::Full(msg) => {
-                                    this.pending_message_to_session = Some(msg);
-                                    // ensure we're woken up again
-                                    cx.waker().wake_by_ref();
-                                    break 'receive
-                                }
-                                TrySendError::Closed(_) => {}
-                            }
+                    match to_session_manager_poll_tx.poll_reserve(cx) {
+                        Poll::Ready(Ok(_)) => {
+                            let _ = to_session_manager_poll_tx.send_item(msg);
                         }
-                    }
+                        Poll::Ready(Err(_)) => {}
```

Comment:
the error cases mean that the channel is closed, which should not be possible but we should return Poll::Ready in that case

Reply:
Oh, really? The previous logic was the following so I was copying that:

```rust
TrySendError::Closed(_) => {}
```

i.e. continue on in the loop. Maybe I'm misreading. Happy to fix just want to clarify!